### PR TITLE
Add update checker

### PR DESCRIPTION
**Pull request by KevinDaGame**
An OP will get a message on join if there is a new update

### CODE REVIEW

File: VoxelSniperCore/src/main/java/com/github/kevindagame/util/VersionChecker.kt [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+     * @param currentVersion The current version of VoxelSniper-Reimagined.
+     *
+     * @return The latest version info, or null if the current version is the latest.
+     */
+    fun getLatestVersionInfo(currentVersion: String): VersionInfo? {
+        val releases = getReleasesFromApi()
+        if (releases.isEmpty()) {
+            return null
+        }
+        val latestRelease = releases[0]
```
Comment (Lennart99) [inline VoxelSniperCore/src/main/java/com/github/kevindagame/util/VersionChecker.kt:26]: Could we assume the first one is the latest version?

File: VoxelSniperCore/src/main/java/com/github/kevindagame/util/VersionChecker.kt [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+    /**
+     * Parses a version string into a Version object.
+     * @param version The version string.
+     *
+     * @return The parsed version.
+     */
+    private fun parseVersion(version: String): Version {
+        val versionParts = version.removePrefix("v").split(".")
```
Comment (Lennart99) [inline VoxelSniperCore/src/main/java/com/github/kevindagame/util/VersionChecker.kt:74]: Might be good to make it lowercase before removing the prefix
```suggestion
        val versionParts = version.lowercase().removePrefix("v").split(".")
```

**Review by Lennart99 — APPROVED** [review]
Few small comments, overall looks good

File: VoxelSniperCore/src/main/java/com/github/kevindagame/util/VersionChecker.kt (hunk 1 above)
Comment (KevinDaGame) [reply VoxelSniperCore/src/main/java/com/github/kevindagame/util/VersionChecker.kt:26]: Good point, The api docs don't explicitly state this, though it seems to be based on publish date, this did make me think of something though, prereleases shouldn't trigger the update warning

File: VoxelSniperCore/src/main/java/com/github/kevindagame/util/VersionChecker.kt (hunk 2 above)
Comment (Lennart99) [reply VoxelSniperCore/src/main/java/com/github/kevindagame/util/VersionChecker.kt:83]: Why did you remove this again?

**Review by Lennart99 — APPROVED** [review]

**Pull request merged**